src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java

```diff
@@ -0,0 +1,94 @@
+package io.jenkins.plugins.analysis.core.columns;
+
+import static io.jenkins.plugins.analysis.core.testutil.Assertions.assertThat;
+
+import io.jenkins.plugins.analysis.warnings.Java;
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.DomNode;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import jenkins.model.Jenkins;
+
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerTest;
+
+/**
+ * Integration Test for the Issue counter column.
+ */
+public class IssueCounterITest extends IntegrationTestWithJenkinsPerTest {
+
+    private static final String ISSUE_ELEMENT_XPATH = "/html/body/div[4]/div[2]/div[2]/div[2]/table/tbody/tr[2]/td[8]/a";
+    private static final String JAVAC_ONE_WARNING = "javac_1_warning.txt";
+    private static final String JAVAC_TEN_WARNINGS = "javac_10_warnings.txt";
+    private static final String RESULT_FILE_NAME = "javac.log";
+    private static final String JAVA_FILE_PATTERN = "**/*.log";
+
+    /**
+     * Tests that the issue counter is displayed correctly with 1 issue.
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToOne() throws IOException {
+        createAndBuildProjectWithFile(JAVAC_ONE_WARNING);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
+
+        assertThat(Integer.parseInt(element.get(0).asText())).isEqualTo(1);
+    }
+
+    /**
+     * Tests that the issue counter is displayed correctly with 10 issues.
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToTen() throws IOException {
+        createAndBuildProjectWithFile(JAVAC_TEN_WARNINGS);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
+
+        assertThat(Integer.parseInt(element.get(0).asText())).isEqualTo(10);
+    }
+
+    /**
+     * Tests that the issue counter is displayed correctly with no issues.
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToZero() throws IOException {
+        FreeStyleProject project = createFreeStyleProject();
+
+        enableWarnings(project, new Java());
+        scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
```

Comment:
Is there a way to check for the health report?

Reply:
I'm not sure what the goal is if this test is for the issue counter column. Should I check for the health status as well? 
Do you mean the hover menu for different tools? If so, I can access it by XPath as well but it might not be stable.

Reply:
Yes, I mean the hover, sorry, not the health report. 
It should be part of the DOM, or isn't it?